include/device.h

```diff
@@ -783,23 +796,31 @@ BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
 		};
 
 #ifdef CONFIG_PM_DEVICE
+#define Z_DEVICE_DEFINE_PM_STRUCT(dev_name) \
+	static struct pm_device Z_PM_DEVICE_NAME(dev_name);
 #define Z_DEVICE_DEFINE_PM_INIT(dev_name, pm_control_fn)		\
-	.pm_control = (pm_control_fn),					\
-	.pm = &Z_DEVICE_STATE_NAME(dev_name).pm,
+	.pm_control = (pm_control_fn),				\
+	.pm = &Z_PM_DEVICE_NAME(dev_name),
 #else
+#define Z_DEVICE_DEFINE_PM_STRUCT(dev_name)
 #define Z_DEVICE_DEFINE_PM_INIT(dev_name, pm_control_fn)
 #endif
 
-#define Z_DEVICE_DEFINE_INIT(node_id, dev_name, pm_control_fn)		\
+#define Z_DEVICE_DEFINE_INIT(node_id, dev_name, pm_control_fn, use_pm)	\
 		.handles = Z_DEVICE_HANDLE_NAME(node_id, dev_name),	\
-		Z_DEVICE_DEFINE_PM_INIT(dev_name, pm_control_fn)
+		COND_CODE_1(use_pm, 					\
+			(Z_DEVICE_DEFINE_PM_INIT(dev_name, pm_control_fn)), \
+			())
```

Comment:
I think that these lines are causing the issue, I think that if `use_mp` is false you need to set `pm_control = NULL` explicitly (might be wrong).